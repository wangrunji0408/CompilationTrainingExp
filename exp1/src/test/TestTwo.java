package test;

public class TestTwo {
    public static int frobble(int x) {
	return 8-x;
    }

    public static void main (String [] args) {
	int a=0, b=0, c=0, d;

	for (a=0; a<7; a++) {

	    for (d=0; d<7; d++) {
	        c=frobble(c);
	    }
				
	    if (a-b == 2) {
		c++;
	    } else if (c == 4) {
		break;
	    }
	}

	c++;

    }
}
