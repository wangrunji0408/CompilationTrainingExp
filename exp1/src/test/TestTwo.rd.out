Now analyzing test.TestTwo
Method: frobble
entry: []
1 in:  []
1 out: [1]
2 in:  [1]
2 out: [1]
exit: [1]
Method: main
entry: []
1 in:  []
1 out: [1]
2 in:  [1]
2 out: [1, 2]
3 in:  [1, 2]
3 out: [1, 2, 3]
4 in:  [1, 2, 3]
4 out: [2, 3, 4]
23 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
23 out: [2, 3, 4, 25, 39, 40, 42, 47, 50]
25 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
25 out: [2, 3, 4, 25, 39, 42, 47, 50]
26 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
26 out: [2, 4, 25, 26, 40, 42, 50]
27 in:  [2, 4, 25, 26, 40, 42, 50]
27 out: [2, 4, 25, 26, 40, 42, 50]
34 in:  [2, 3, 4, 25, 38, 39, 40, 42, 47, 50]
34 out: [2, 3, 4, 25, 38, 39, 40, 42, 47, 50]
38 in:  [2, 3, 4, 25, 38, 39, 40, 42, 47, 50]
38 out: [2, 3, 4, 25, 38, 39, 40, 47, 50]
39 in:  [2, 3, 4, 25, 38, 39, 40, 47, 50]
39 out: [2, 4, 25, 38, 39, 40, 50]
40 in:  [2, 4, 25, 38, 39, 40, 50]
40 out: [2, 4, 38, 39, 40, 50]
41 in:  [2, 4, 38, 39, 40, 50]
41 out: [2, 4, 38, 39, 40, 50]
42 in:  [2, 3, 4, 25, 38, 39, 40, 42, 47, 50]
42 out: [2, 3, 4, 25, 39, 40, 42, 47, 50]
43 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
43 out: [2, 3, 4, 25, 39, 40, 42, 47, 50]
47 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
47 out: [2, 4, 25, 40, 42, 47, 50]
48 in:  [2, 4, 25, 40, 42, 47, 50]
48 out: [2, 4, 25, 40, 42, 47, 50]
49 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
49 out: [2, 3, 4, 25, 39, 40, 42, 47, 50]
50 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
50 out: [2, 3, 25, 39, 40, 42, 47, 50]
51 in:  [2, 3, 25, 39, 40, 42, 47, 50]
51 out: [2, 3, 25, 39, 40, 42, 47, 50]
52 in:  [2, 3, 4, 25, 39, 40, 42, 47, 50]
52 out: [2, 3, 4, 25, 39, 40, 42, 47, 50]
exit: [2, 3, 4, 25, 26, 38, 39, 40, 42, 47, 50]
Method: <init>
entry: []
1 in:  [2]
1 out: [2]
2 in:  []
2 out: [2]
3 in:  [2]
3 out: [2]
exit: [2]
